src/bin/sample_range.rs

```diff
@@ -145,38 +144,32 @@ async fn audit_block(
     let receipts_ck = HistoryContentKey::BlockReceipts(BlockReceiptsKey {
         block_hash: hash.to_fixed_bytes(),
     });
-    match client.recursive_find_content(header_ck).await? {
-        ContentInfo::Content { content, .. } => {
-            if let PossibleHistoryContentValue::ContentPresent(_) = content {
-                metrics.lock().unwrap().header.success_count += 1;
-            } else {
-                warn!("Header not found for block {hash}");
-                metrics.lock().unwrap().header.failure_count += 1;
-            }
+    match client.recursive_find_content(header_ck).await {
+        Ok(_) => {
```

Comment:
Doesn't this change the logic?
It could be some other kind of `ContentInfo` and we would count it as success, while previously we wouldn't.
Or am I missing something :thinking:

Reply:
The update in this pr is to return an `Error` in the case where a content lookup failed. In this script, all we really care about is whether the content lookup succeeded/failed, eg. we don't really care about the actual piece of content. Previously, we had to unpack the `ContentInfo` to make sure that it wasn't the "possible content value" variant. Now, all `ContentInfo`s are expected to be valid pieces of content, and all failed content lookups result in an `Error`, which we handle below.